services/statestorage/test/get_state_hash.go

```diff
@@ -0,0 +1,39 @@
+package test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+)
+
+var _ bool = Describe("Getting Merkle Root", func() {
+	When("invoking method", func() {
+		It("returns a hash code", func() {
+			d := newStateStorageDriver(1)
+
+			root, err := d.service.GetStateHash(&services.GetStateHashInput{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(root.StateRootHash)).ToNot(BeZero())
+		})
+	})
+
+	When("changing state", func() {
+		When("adding first key/value", func() {
+			It("merkle root changes", func() {
+				d := newStateStorageDriver(1)
+
+				root1, err := d.service.GetStateHash(&services.GetStateHashInput{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(len(root1.StateRootHash)).ToNot(BeZero())
+
+				d.commitValuePairs("foo", "bar", "baz")
```

Comment:
Doesn't commit need to specify the block height? How will we commit in a test the keys for the next block?

Reply:
this is a test driver. it conveniently commits at the next block height. 
there are corresponding methods for committing at specific block heights